src/tree/registries/azure/AzureAccountTreeItem.ts

```diff
@@ -28,4 +29,18 @@ export class AzureAccountTreeItem extends AzureAccountTreeItemBase implements IR
     public createSubscriptionTreeItem(subContext: ISubscriptionContext): SubscriptionTreeItem {
         return new SubscriptionTreeItem(this, subContext);
     }
+
+    public async loadMoreChildrenImpl(_clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        const treeItems: AzExtTreeItem[] = await super.loadMoreChildrenImpl(_clearCache, context);
+
+        if (treeItems.length === 1 && treeItems[0].commandId === 'extension.open') {
+            const extensionUtils: ExtensionUtils = new ExtensionUtils('ms-vscode.azure-account');
```

Comment:
This will still create `n` numbers of listeners of extension events, for `n` refreshes of this node or parent node, when there should just be, at most, one listener.  One of the tricks will be understanding when a node no longer needs to listen; the easy case is when the same instance is refreshed and the waited-for extension is now installed, but the hard case is when the node itself is thrown away in favor of a brand new node, as I don't think there's a disposal pattern in place for tree view nodes.

That said, I think there should only ever be just *one* of this type of node in place at any given time, so we might use construction of a new node to indicate that the existing node should stop listening (as, presumably, it's being thrown away).

Reply:
When the node gets refreshes, the same instance is not used instead the old instance is thrown away. AzureAccountTreeItemBase which is the base class for this type implements disposable. But unfortunately when the node/parent gets refreshed, the dispose is not getting called, so can't  use the dispose() method to dispose the listener.

Reply:
Now listening only once.